samples/nrf9160/fmfu_smp_svr/src/main.c

```diff
@@ -7,9 +7,41 @@
 #include <zephyr.h>
 #include <mgmt/fmfu_mgmt.h>
 #include <mgmt/fmfu_mgmt_stat.h>
+#include <nrf_modem_full_dfu.h>
 #include <modem/modem_info.h>
 #include <modem/nrf_modem_lib.h>
 #include <modem/at_cmd.h>
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include "img_mgmt/img_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+#include "os_mgmt/os_mgmt.h"
+#endif
+
+static void fmfu_callback_handler(const enum fmfu_evt_id id)
+{
+	switch (id) {
+	case FMFU_EVT_INIT:
+		/* Shutdown modem to prepare for DFU */
+		nrf_modem_lib_shutdown();
```

Comment:
Could it be an option to have a separate function inside `fmfu_mgmt` which performs these steps instead of requiring the user to invoke these? That way if the API towards `nrf_modem` changes those changes wouldn't be visible for the end user.

```
case FMFU_EVT_INIT:
     /* Do whatever the app needs to do */
     err = fmfu_mgmt_start();
     /* ... */
```

Reply:
Had the same idea but then we will hide from the user what is actually happening which is shutting down the modem library. I guess it would be reasonable to assume that they won't use the socket API etc while trying to do this.

Maybe provided a `fmfu_mgmt_default_init()`  function could be an idea so that they know they can fill in what is needed to be done.

Reply:
Good point, since we state that we need special handling I guess they will take extra care setting this up. Maybe this is something we can describe a bit in the .rst ?

IMO we should either take full ownership of the procedure, or leave the whole procedure to the user.